diff --git a/Cargo.toml b/Cargo.toml
index dea0d60..8b45b1b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ readme = "README.md"

 [lib]
 name = "etebase"
-crate-type = ["cdylib"]
+crate-type = ["cdylib", "staticlib"]

 [build-dependencies]
 cbindgen = "^0.14.2"
diff --git a/Makefile b/Makefile
index 8b7f1af..a6a27ed 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ DST_PKGCONFIG_DIR = $(DST_LIBRARY_DIR)/pkgconfig
 DST_CMAKECONFIG_DIR = $(DST_LIBRARY_DIR)/cmake/Etebase
 DST_INCLUDE_DIR = $(DESTDIR)$(PREFIX)/include/$(PKGNAME)

-LIBRARY_FILE = $(BUILDDIR)/$(MODE)/lib$(PKGNAME).so
+LIBRARY_FILE = $(BUILDDIR)/$(MODE)/lib$(PKGNAME).dylib
 HEADER_FILE = $(BUILDDIR)/$(PKGNAME).h
 PKGCONFIG_FILE = $(BUILDDIR)/$(PKGNAME).pc
 CMAKECONFIG_FILE = EtebaseConfig.cmake
diff --git a/build.rs b/build.rs
index 69a99ca..43d1d19 100644
--- a/build.rs
+++ b/build.rs
@@ -7,7 +7,7 @@ use std::env;

 fn main() {
     let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    if cfg!(unix) && !cfg!(macos) {
+    if cfg!(unix) && cfg!(not(target_os="macos")) {
         // Set soname for platforms other than Windows
         const VERSION_MAJOR: &str = env!("CARGO_PKG_VERSION_MAJOR");
         println!(
