diff --git a/build.rs b/build.rs
index 601baab..5a1c797 100644
--- a/build.rs
+++ b/build.rs
@@ -7,11 +7,6 @@ use std::env;
 
 fn main() {
     let crate_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    if cfg!(unix) {
-        // Set soname for platforms other than Windows
-        const VERSION_MAJOR: &'static str = env!("CARGO_PKG_VERSION_MAJOR");
-        println!("cargo:rustc-cdylib-link-arg=-Wl,-soname,libetebase.so.{}", VERSION_MAJOR);
-    }
     match cbindgen::generate(&crate_dir) {
         Ok(gen) => gen,
         Err(e) => match e {
